zrcs/zrc-2.md

```diff
@@ -0,0 +1,412 @@
+
+|           | Name             | Description                 | Event Parameters                                                                                                                                                                                                                                              |
+| --------- | ---------------- | --------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| eventName | `OperatorBurnSuccess` | Burning is successful.     | `operator`: `ByStr20`, `tokenOwner`: `ByStr20`, and `amount`: `Uint128`.                                                                                               |
+| eventName | `Error`          | Burning is not successful. | - emit `CodeNotAuthorised` if the transition is called by an operator who is not authorized. |
+
+
+#### 7. Mint
+
+```ocaml
+(* @dev: Mint new tokens. Only contractOwner can mint.                        *)
+(* @param recipient:     Address of the recipient whose balance is increased. *)
+(* @param amount:        Number of tokens to be burned.                       *)
+transition Mint(recipient: ByStr20, amount: Uint128)
+```
+
+|        | Name        | Type      | Description                                          |
+| ------ | ----------- | --------- | ---------------------------------------------------- |
+| @param | `recipient` | `ByStr20` | Address of the recipient whose balance is increased. |
+| @param | `amount`    | `Uint128` | Number of tokens to be minted.                       |
+
+|           | Name          | Description                | Event Parameters                                                                                                                                                                                                                   |
+| --------- | ------------- | -------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| eventName | `MintSuccess` | Minting is successful.     | `recipient`: `ByStr20`, and `amount`: `Uint128`.                             |
+| eventName | `Error`       | Minting is not successful. | - emit `CodeTokenExists` if the token already exists.<br>- emit `CodeNotAuthorised` if the transition is called by a user who is not the contract owner.<br>**NOTE:** Only the `contractOwner` is allowed to call this transition. |
+
+
+#### 8. OperatorMint
+
+```ocaml
+(* @dev: Mint new tokens. Only approved operator can mint tokens.         *)
+(* @param operator:   Address must be an operator of tokenOwner.          *)
```

Comment:
Similar to `OperatorSend`, the `operator` need not be a parameter here. It is just the `_sender`.